src/lib.rs

```diff
@@ -64,20 +64,63 @@ pub mod raftstore;
 pub mod server;
 pub mod storage;
 
+// Detect tikv enabled features.
+fn detect_enabled_features() -> String {
+    macro_rules! detect {
+        ($f:expr, $feat:tt) => {
+            if cfg!(feature = $feat) {
+                $f.push_str($feat);
+                $f.push(' ');
+            }
+        };
+    }
+
+    let mut flags = String::new();
+    detect!(flags, "tcmalloc");
+    detect!(flags, "jemalloc");
+    detect!(flags, "mimalloc");
+    detect!(flags, "portable");
+    detect!(flags, "sse");
+    detect!(flags, "mem-profiling");
+    detect!(flags, "failpoints");
+    detect!(flags, "prost-codec");
+    detect!(flags, "protobuf-codec");
+    flags.pop(); // Try to trim the last ' '.
+    flags
+}
+
+// Detect tikv build profile.
+fn detect_build_profile() -> &'static str {
+    // Detect build profile by debug_assertions see more:
+    // https://users.rust-lang.org/t/conditional-compilation-for-debug-release/1098
+    // TODO: Find a better way to detect, since debug_assertions can be enabled
+    //       in a custom profile.
+    if cfg!(debug_assertions) {
```

Comment:
You can use the `PROFILE` env var rather than doing this.

Reply:
I thought about it, but the `PROFILE` only available in build script. And build script may not be a good option, because it invalidates build-cache incorrectly, see more #3368 .